usr/src/uts/common/fs/zfs/range_tree.c

```diff
@@ -343,6 +357,15 @@ range_tree_contains(range_tree_t *rt, uint64_t start, uint64_t size)
 }
 
 /*
+ * Same as range_tree_contains, but locates even just a partial overlap.
+ */
+boolean_t
+range_tree_contains_part(range_tree_t *rt, uint64_t start, uint64_t size)
+{
+	return (range_tree_find_impl(rt, start, size) != NULL);
```

Comment:
Yes, the caller needs to hold the lock already. Again, the same logic of making locking external and explicit applies here. This function is used in two spots: metaslab_trim_add and metaslab_exec_trim. The latter function contains an ASSERT to make sure the lock is held. The former does not. I can add an explicit ASSERT(MUTEX_HELD()) to metaslab_trim_add.